packages/flutter_tools/lib/src/web/file_generators/js/flutter.js

```diff
@@ -156,17 +156,15 @@ _flutter.loader = null;
     }
 
     /**
-     * Returns the latest service worker for the given `serviceWorkerRegistrationPromise`.
+     * Returns the latest service worker for the given `ServiceWorkerRegistration`.
      *
      * This might return the current service worker, if there's no new service worker
      * awaiting to be installed/updated.
      *
-     * @param {Promise<ServiceWorkerRegistration>} serviceWorkerRegistrationPromise
-     * @returns {Promise<ServiceWorker>}
+     * @param {ServiceWorkerRegistration} reg
+     * @returns {ServiceWorker}
      */
-    async _getNewServiceWorker(serviceWorkerRegistrationPromise) {
-      const reg = await serviceWorkerRegistrationPromise;
-
+    _getNewServiceWorker(reg) {
```

Comment:
Can you substitute a more descriptive name

```suggestion
    _getNewServiceWorker(serviceWorkerRegistration) {
```

Reply:
Yeah, I just didn't want to rename existing usages of the `reg` variable below

Reply:
please do

Reply:
Should I also rename `serviceWorker` to the `latestServiceWorker` in  `_waitForServiceWorkerActivation` function?    
I don't like that this function currently has meaningless `const serviceWorker = latestServiceWorker;` line

Reply:
Or it can be the other way, `latestServiceWorker` -> `serviceWorker`

Reply:
oh yeah, we don't need that extra assignment now that we don't have an await.